docs/tuning-apps/mpi-io/ompio.rst

```diff
@@ -0,0 +1,262 @@
+
+#. ``io_ompio_num_aggregators``: Number of aggregators used in
+   collective I/O operations.  Setting this parameter to a value
+   larger zero disables the internal automatic aggregator selection
+   logic of OMPIO.  Tuning this parameter has a very high impact on
+   the performance of collective operations.
+
+   .. note:: Be sure to also see recommendations for tuning collective
+             operations.
+
+#. ``io_ompio_grouping_option``: Algorithm used to automatically
+   decide the number of aggregators used. Applications working with
+   regular 2-D or 3-D data decomposition can try changing this
+   parameter to 4 (hybrid) algorithm.
+
+#. ``fs_ufs_lock_algorithm``: Parameter used to determing what part of
+   a file needs to be locked for a file operation. Since the ``ufs``
+   ``fs`` component is used on multiple file systems, OMPIO
+   automatically chooses the value required for correctness on all
+   file systems, e.g. enforcing locking on an NFS file system, while
+   disabling locking on a local file system. Users can adjust the
+   required locking behavior based on their use case, since the
+   default value might often be too restrictive for their application.
+
+Setting stripe size and stripe width on parallel file systems
+-------------------------------------------------------------
+
+Many ``fs`` components allow you to manipulate the layout of a new
+file on a parallel file system.  Note, that many file systems only
+allow changing these setting upon file creation, i.e. modifying these
+values for an already existing file might not be possible.
+
+#. ``fs_pvfs2_stripe_size``: Sets the number of storage servers for a
+   new file on a PVFS2/OrangeFS  file system. If not set, system default will be
+   used. Note that this parameter can also be set through the
+   ``stripe_size`` MPI Info value.
+
+#. ``fs_pvfs2_stripe_width``: Sets the size of an individual block for
+   a new file on a PVFS2 file system. If not set, system default will
+   be used. Note that this parameter can also be set through the
+   ``stripe_width`` MPI Info value.
+
+#. ``fs_lustre_stripe_size``: Sets the number of storage servers for a
+   new file on a Lustre file system. If not set, system default will
+   be used. Note that this parameter can also be set through the
+   ``stripe_size`` MPI Info value.
+
+#. ``fs_lustre_stripe_width``: Sets the size of an individual block
+   for a new file on a Lustre file system. If not set, system default
+   will be used. Note that this parameter can also be set through the
+   ``stripe_width`` MPI Info value.
+
+Using GPU device buffers in MPI File I/O operations
+----------------------------------------------------
+
+OMPIO supports reading and writing directly to/from GPU buffers using
+the MPI I/O interfaces. Using this feature simplifies managing buffers
+that are exclusive used on GPU devices, and hence the necessity to
+implement a staging through host memory for file I/O operations.
+
```

Comment:
proof-reading: '... that are exclusively used on GPU devices, and hence there is no need to implement staging through host memory for file I/O operations'